app/src/main/java/protect/card_locker/preferences/SettingsActivity.java

```diff
@@ -144,8 +148,44 @@ public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
                 }
             }
             localePreference.setEntries(entries.toArray(new CharSequence[entryValues.length]));
+            //make locale picker preference in sync with system settings
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                Locale chosenLocale = AppCompatDelegate.getApplicationLocales().get(0);
+                if (chosenLocale == null) {
+                    //corresponds to "System"
+                    localePreference.setValue("");
+                } else {
+                    //need to set preference's value to one of localePreference.getEntryValues() to match the locale.
+                    //Locale.toLanguageTag() theoretically should be one of the values in localePreference.getEntryValues()...
+                    //but it doesn't work for some locales. so trying something more heavyweight.
+
+                    //Obtain all locales supported by the app.
+                    List<Locale> supportedLocales = Arrays.stream(localePreference.getEntryValues())
+                            .map(Objects::toString)
+                            .map(Utils::stringToLocale)
+                            .collect(Collectors.toList());
+                    //Find the locale that's selected in settings and get its index in supported locales
+                    for (int i = 0; i < supportedLocales.size(); i++) {
+                        Locale localeToCompare = supportedLocales.get(i);
+                        //If found, set preference value to entry value at that index
+                        //Android 13 settings seems to "force" the user to select country of locale, but many app-supported locales only have language, not country
+                        if (Utils.localeEqualsAfterAdjust(localeToCompare, chosenLocale)) {
+                            localePreference.setValue(localePreference.getEntryValues()[i].toString());
+                            break;
+                        }
+                    }
```

Comment:
Actually, I noticed one last bug only on Android 13 and 14 (aside from that, this code severely improves language handling):

If you specifically choose "español (Argentina)" in Catima's settings, Catima will force it to "español". I'm marking this code as this seems to be the most logical part where that bug could happen, but I may be wrong as I haven't debugged deeply.

Reply:
Yeah, the bug is there. Will see if I can fix.

Reply:
It will match the first language that's "equal", which is "español". It never sees that "español (Argentina)" is a closer match. See patch below for a possible fix.

Reply:
Double tested this just now and it seems to work well :)